47.md

```diff
@@ -63,6 +63,267 @@ Response:
 }
 ```
 
+Below are the commands that are defined in this NIP.
+
+### Pay Invoice
+
+Request:
+```jsonc
+{
+    "method": "pay_invoice",
+    "params": {
+        "invoice": "lnbc50n1..." // invoice to pay, required
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "preimage": "0123456789abcdef..." // preimage of the payment
+    }
+}
+```
+
+### Keysend
+
+Request:
+```jsonc
+{
+    "method": "pay_keysend",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "pubkey": "03...", // payee pubkey, required
+        "message": "some message", // keysend message, optional
+        "preimage": "0123456789abcdef...", // preimage of the payment, optional
+        "tlv_records: [ // tlv records, optional
+            {
+                "type": 5482373484, // tlv type
+                "value": "fajsn341414fq" // hex encoded tlv value
+            }
+        ]
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_keysend",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+        "payment_hash": "31afdf1..", // payment hash of the payment
```

Comment:
if we return the payment_hash here should `pay_invoice` also return the payment hash? 
just thinking about consistency even though it is not needed there as the user has the invoice anyway.

Reply:
i just removed the `payment_hash` to be consistent, either way it is derivable from the preimage